tee-worker/litentry/core/vc-issuance/lc-vc-task-receiver/src/lib.rs

```diff
@@ -119,104 +121,108 @@ where
 		.try_into()
 		.map_err(|e| format!("Failed to convert to UserShieldingKeyType: {:?}", e))?;
 
-	let decrypted_trusted_operation = match aes_decrypt(&aes_key, &mut encrypted_trusted_call) {
-		Some(s) => s,
-		None => return Err("Failed to decrypted trusted operation".to_string()),
-	};
+	let decrypted_trusted_operation = aes_decrypt(&aes_key, &mut encrypted_trusted_call)
+		.ok_or_else(|| "Failed to decrypt trusted operation".to_string())?;
 
 	let trusted_operation = TrustedOperation::<TrustedCallSigned, Getter>::decode(
 		&mut decrypted_trusted_operation.as_slice(),
 	)
 	.map_err(|e| format!("Failed to decode trusted operation, {:?}", e))?;
 
-	let trusted_call: &TrustedCallSigned = match trusted_operation.to_call() {
-		Some(s) => s,
-		None => return Err("Failed to convert trusted operation to trusted call".to_string()),
-	};
+	let trusted_call: &TrustedCallSigned = trusted_operation
+		.to_call()
+		.ok_or_else(|| "Failed to convert trusted operation to trusted call".to_string())?;
 
 	if let TrustedCall::request_vc(signer, who, assertion, maybe_key, _hash) =
 		trusted_call.call.clone()
 	{
-		let (mut state, _) = context.state_handler.load_cloned(&shard).map_err(|e| {
-			format!("Received error while trying to obtain sidechain state: {:?}", e)
-		})?;
-
-		state.execute_with(|| {
-			let key = match maybe_key {
-				Some(s) => s,
-				None => return Err("User shielding key not provided".to_string()),
-			};
-
-			let id_graph = IMT::get_id_graph(&who);
-			let assertion_networks = assertion.clone().get_supported_web3networks();
-			let identities: Vec<IdentityNetworkTuple> = id_graph
-				.into_iter()
-				.filter(|item| item.1.is_active())
-				.map(|item| {
-					let mut networks = item.1.web3networks.to_vec();
-					networks.retain(|n| assertion_networks.contains(n));
-					(item.0, networks)
-				})
-				.collect();
-
-			let signer = match signer.to_account_id() {
+		let key = maybe_key.ok_or_else(|| "User shielding key not provided".to_string())?;
+		let identities: Vec<IdentityNetworkTuple> = context
+			.state_handler
+			.execute_on_current(&shard, |state, _| {
+				let prefix_key = storage_map_key(
+					"IdentityManagement",
```

Comment:
Can you remove this hardcoded values and rely on generated ones ?

```
					pallet_identity_management_tee::pallet::IDGraphs::<Runtime>::module_prefix(),
					pallet_identity_management_tee::pallet::IDGraphs::<Runtime>::storage_prefix(),
```

Reply:
Hmm maybe this can be further simplified to this:

```
				let prefix_key = pallet_identity_management_tee::pallet::IDGraphs::<Runtime>::storage_double_map_final_key1(&who);
```

Reply:
I am slightly confused here, 
`module_prefix()` and `storage_prefix()` are defined for `IDGraphs` ? I don't find these functions anywhere could you share me some documentation I can look up at? And can these functions be called outside of the externalities environment?

Reply:
This function is available on trait `frame_support::storage::generator::StorageDoubleMap` . The trait impl is added by `#[pallet::storage]` macro expansion.


The trait itself is hidden from docs so you need to look at frame-support source code to find out.

```
#[doc(hidden)]
pub mod generator;
```

It's documented as follows:
```
	/// Generate the first part of the key used in top storage.
	fn storage_double_map_final_key1<KArg1>(k1: KArg1) -> Vec<u8>
```

Reply:
It compiles, but I get the following error when I perform e2e test: 
```
thread '<unnamed>' panicked at 'storage cannot be called outside of an Externalities-provided environment.', /home/faisal/litentry-parachain/tee-worker/core-primitives/substrate-sgx/sp-io/src/lib.rs:131:9
```
As I mentioned I don't think we can call this outside of externalities

Reply:
You are right, I though it's just a static string - don't know yet why it fails 😿